src/js/closable-surveys.js

```diff
@@ -0,0 +1,89 @@
+import LocalStorageManager from './localStorageManager';
+
+// eslint-disable-next-line func-names
+(function (Drupal) {
+  Drupal.behaviors.closable_surveys = {
+    attach: function attach() {
+      const survey = document.getElementById('helfi-survey__container');
+      if (!survey) return;
+
+      const root = document.documentElement;
+      const surveyDelay = 2000;
+      const surveyCloseButton = document.getElementById('helfi-survey__close-button');
+      const surveyKey = 'hidden-helfi-surveys';
+      const storageManager = new LocalStorageManager('helfi-settings');
+      let surveysToHide = null;
+      let surveyFocusTrap = null;
+
+      try {
+        surveysToHide = JSON.parse(window.localStorage.getItem('helfi-settings'));
+      } catch (e) {
+        console.error('Error parsing local storage data:', e);
+      }
+
+      function addToSurveyStorage() {
+        const { uuid } = survey.dataset;
+        if (uuid) {
+          storageManager.addValue(surveyKey, uuid);
+        }
+      }
+
+      function addFocusTrap() {
+        surveyFocusTrap = window.focusTrap.createFocusTrap('#helfi-survey', {
+          initialFocus: () => '#helfi-survey__title'
+        });
+        surveyFocusTrap.activate();
+      }
+
+      function toggleNoScroll(enable) {
+        root.classList.toggle('noscroll', enable);
+      }
+
+      function setBodyPaddingRight(enable) {
+        if (enable) {
+          const scrollBarWidth = `${window.innerWidth - document.documentElement.clientWidth}px`;
+          document.body.style.paddingRight = scrollBarWidth;
+        } else {
+          document.body.style.removeProperty('padding-right');
+        }
+      }
+
+      function showSurvey() {
+        const { uuid } = survey.dataset;
+        const hiddenSurveys = surveysToHide?.[surveyKey] || [];
+        const shouldShowSurvey = !uuid || !hiddenSurveys.includes(uuid);
+
+        if (shouldShowSurvey) {
+          survey.style.display = 'flex';
+          toggleNoScroll(true);
+          setBodyPaddingRight(true);
```

Comment:
This should be moved above "toggleNoScroll(true)", otherwise some browsers return "0px" for the padding value.

Reply:
This should be fixed now 👍 Nicely noticed 💪